cmd/jujud/agent/controllercharm.go

```diff
@@ -0,0 +1,265 @@
+// Copyright 2021 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package agent
```

Comment:
Should this be in it's own package as theoretically it doesn't need to be tied to the bootstrap process.

Reply:
As with the step to download and unpack the dashboard (done just before the controller charm set up), this is part of bootstrap in that it needs to be done as part of setting up the controller. It imports apiserver, state, environ and so needs to be done in a place where we don't run into import loops or break layering etc.